server/server.go

```diff
@@ -103,8 +120,20 @@ func (s *RPCServer) AddMiddleware(m MiddlewareFunc) *RPCServer {
 }
 
 // AddHandler adds a new handler to the RPC server.
-func (s *RPCServer) AddHandler(queueName string, handler HandlerFunc) {
-	s.handlers[queueName] = handler
+func (s *RPCServer) AddHandler(routingKey string, handler HandlerFunc) {
+	s.AddExchangeHandler(routingKey, "", "direct", amqp.Table{}, handler)
+}
+
+func (s *RPCServer) AddExchangeHandler(routingKey, exchangeName, exchangeType string, bindHeaders amqp.Table, handler HandlerFunc) {
```

Comment:
Should we take a struct here instead?

That way we can prepare some pre-defined ones:

```go
server.AddExhangeHandler(DirectBinding("my.queue.name"), handler)
```